x-pack/test/functional/services/ml/data_visualizer_index_based.ts

```diff
@@ -231,5 +232,84 @@ export function MachineLearningDataVisualizerIndexBasedProvider({
         }
       );
     },
+
+    async assertRandomSamplingOptionsButtonExists() {
+      await testSubjects.existOrFail('dvRandomSamplerOptionsButton');
+    },
+
+    async assertRandomSamplingOption(
+      expectedOption:
+        | 'dvRandomSamplerOptionOnAutomatic'
+        | 'dvRandomSamplerOptionOnManual'
+        | 'dvRandomSamplerOptionOff',
+      expectedProbability?: number
+    ) {
+      await retry.tryForTime(2000, async () => {
+        await browser.pressKeys(browser.keys.ESCAPE);
+        await testSubjects.clickWhenNotDisabled('dvRandomSamplerOptionsButton');
+        await testSubjects.existOrFail('dvRandomSamplerOptionsPopover');
+
+        if (expectedOption === 'dvRandomSamplerOptionOff') {
+          await testSubjects.existOrFail('dvRandomSamplerOptionOff');
+          await testSubjects.missingOrFail('dvRandomSamplerProbabilityRange');
+          await testSubjects.missingOrFail('dvRandomSamplerAutomaticProbabilityMsg');
+        }
+
+        if (expectedOption === 'dvRandomSamplerOptionOnManual') {
+          await testSubjects.existOrFail('dvRandomSamplerOptionOnManual');
+          await testSubjects.existOrFail('dvRandomSamplerProbabilityRange');
+          if (expectedProbability !== undefined) {
+            const probability = await testSubjects.getAttribute(
+              'dvRandomSamplerProbabilityRange',
+              'value'
+            );
+            expect(expectedProbability).to.eql(
+              `${probability}`,
```

Comment:
This should be the other way around:
```suggestion
            expect(robability).to.eql(
              `${expectedProbability}`,
```
With the custom assertion message, it's not super bad, but the `expect`-internal failure message that's still printed out would have the actual and expected values wrong.

Reply:
Thanks for catching that! Updated here [`9196446` (#142278)](https://github.com/elastic/kibana/pull/142278/commits/91964466475af8eddaa364bacc027e4b80ae417f)